src/dft.cpp

```diff
@@ -70,6 +72,21 @@ dft_chunk::dft_chunk(fields_chunk *fc_, ivec is_, ivec ie_, vec s0_, vec s1_, ve
   dV0 = dV0_;
   dV1 = dV1_;
 
+  persist = data->persist;
+
+  /* for adjoint calculations, we want to pad
+  (or expand) the dimensions of the dft region
+  to account for boundary effects. We will pad
+  by 1 pixel in each dimension, while ensuring
+  we don't step outside of the chunk loop itself
+  */
+  if(data->expand){
+    is_old = is_;
+    ie_old = ie_;
+    is = max(is-one_ivec(fc->gv.dim)*2,fc->gv.little_corner());
+    ie = min(ie+one_ivec(fc->gv.dim)*2,fc->gv.big_corner());
+  }
```

Comment:
It seems like the issue lies with chunks next to each other within the region. There's no problem with the edge of the dft region, so long as it remains in the same chunk. But the interface *between* chunks is problematic. Here's a poor man's illustration:
![image](https://user-images.githubusercontent.com/24902086/148254834-b86c148a-98f5-4fac-8985-30830e194b74.png)

Specifically, `index()` returns an out of bounds index when it tries to pull the off-diagonal terms need for the inner product.

Ideally, the padding should account for this. But it appears that the edge of the `fields` chunk is preventing us from padding an extra pixel, in these cases (at least that's what it looks like when I print all the relevant values).

Reply:
I reverted some unintentional `iyee_shift` in `dft.cpp`. I no longer see the issue described above, but the gradients are still inaccurate when multiple chunks are used (although the relative error is still pretty low).

I'm not sure the error lies with `loop_in_chunks()` either as I'm not using that in the recombination step (although `loop_in_chunks()` is used while creating the `dft_chunk`s, so that may be the issue).